odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,91 @@
+    _description = "Avatar Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920")
+
+    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024")
+    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512")
+    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256")
+    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128")
+
+    def _compute_avatar(self, avatar_field, image_field):
+        for record in self:
+            avatar = record[image_field]
+            if not avatar:
+                if record[record._rec_name]:
+                    avatar = record._avatar_generate_svg()
+                else:
+                    avatar = record._avatar_get_placeholder()
+            record[avatar_field] = avatar
+
+    @api.depends(lambda self: [self._rec_name, 'image_1920'])
+    def _compute_avatar_1920(self):
+        self._compute_avatar('avatar_1920', 'image_1920')
+
+    @api.depends(lambda self: [self._rec_name, 'image_1024'])
+    def _compute_avatar_1024(self):
+        self._compute_avatar('avatar_1024', 'image_1024')
+
+    @api.depends(lambda self: [self._rec_name, 'image_512'])
+    def _compute_avatar_512(self):
+        self._compute_avatar('avatar_512', 'image_512')
+
+    @api.depends(lambda self: [self._rec_name, 'image_256'])
+    def _compute_avatar_256(self):
+        self._compute_avatar('avatar_256', 'image_256')
+
+    @api.depends(lambda self: [self._rec_name, 'image_128'])
+    def _compute_avatar_128(self):
+        self._compute_avatar('avatar_128', 'image_128')
+
+    def _avatar_get_seed(self):
+        return self[self._rec_name] + str(self.create_date.timestamp())
+
+    def _avatar_generate_svg(self):
+        initial = html_escape(self[self._rec_name][0].upper())
+        hashed_seed = sha512(self._avatar_get_seed().encode('utf-8')).hexdigest()
+        hue = self._avatar_generate_hue(hashed_seed)
+        sat = self._avatar_generate_sat(hashed_seed)
+        lig = self._avatar_generate_lig(hashed_seed)
+        bgcolor = html_escape(f'hsl({hue}, {sat}%, {lig}%)')
+        return b64encode(bytes((
+            "<?xml version='1.0' encoding='UTF-8' ?>"
+            "<svg height='180' width='180' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>"
+            f"<rect fill='{bgcolor}' height='180' width='180'/>"
+            f"<text fill='#ffffff' font-size='96' text-anchor='middle' x='90' y='125' font-family='sans-serif'>{initial}</text>"
+            "</svg>"
+        ), 'utf-8'))
+
+    def _avatar_get_placeholder(self):
+        return b64encode(file_open("base/static/img/avatar_grey.png", 'rb').read())
+
+    def _avatar_generate_hue(self, hashed_seed):
```

Comment:
No need to use intermediate variables, as it adds nothing to comprehension. Could be directly
``return int(value, 16) * 360 / 255`` and maybe add a docstring if you want to explain what you are doing